docs/api/advanced/snapshot.mdx

```diff
@@ -7,22 +7,95 @@ description: 'Create a snapshot of current state'
 
 # snapshot
 
-#### A snapshot takes a proxy and returns a immutable object, unwrapped from the proxy.
+snapshot takes a proxy and returns an immutable object, unwrapped from the proxy.
 
-Immutability is achieved by deeply freezing the object. In sequential snapshot calls, when the values in the proxy have not changed, a pointer to the same object is returned. This allows for shallow comparison in render functions, preventing spurious renders. Snapshots also throw promises, making them work with React Suspense.
+Immutability is achieved by deeply copying & freezing the object. In sequential snapshot calls, when the values in the proxy have not changed, a pointer to the same object is returned. This allows for shallow comparison in render functions, preventing spurious renders. Snapshots also throw promises, making them work with React Suspense.
```

Comment:
we want to emphasize that the copying is optimized and most efficient.

Reply:
Sounds good; added an "efficient" and "see copy on write section later" to hopefully highlight that.